src/utils/profileLink.ts

```diff
@@ -0,0 +1,134 @@
+import * as path from "path";
+import * as fs from "fs";
+import { Profiles } from "../Profiles";
+import { ZoweTreeNode } from "../abstract/ZoweTreeNode";
+import { IZoweTreeNode } from "../api/IZoweTreeNode";
+import { getZoweDir } from "../utils";
+import * as nls from "vscode-nls";
+const localize = nls.config({ messageFormat: nls.MessageFormat.file })();
+
+const LINKS_FOLDER = "profile_links";
+const FILE_SUFFIX = ".yaml";
+
+export async function getLinkedProfile(node: IZoweTreeNode, type: string, logger?: Logger) {
+    try {
+        if (node instanceof ZoweTreeNode) {
+            return findLinkedProfile(await getProfile(node), type);
+        }
+        throw new Error(localize("profileLink.notTreeItem", "Tree Item is not a Zowe Explorer item."));
+    } catch (err) {
+        if (logger) {
+            logger.warn(err.message);
+        }
+        throw (err);
+    }
+}
+
+export function getProfile(node: vscode.TreeItem) {
+    if (node instanceof ZoweTreeNode) {
+        return (node as ZoweTreeNode).getProfile();
+    }
+    throw new Error(localize("profileLink.notTreeItem", "Tree Item is not a Zowe Explorer item."));
+}
+
+export async function linkProfileDialog(aProfile: IProfileLoaded) {
+    let chosenName;
+    let chosenType;
+    if (aProfile) {
+        const possibles = Profiles.getInstance().getAllTypes().filter( (value) => value !== aProfile.type);
+        const quickPickOptions1: vscode.QuickPickOptions = {
+            placeHolder: localize("profileLink.selectAltProfile", "Select a type of alternative profile to associate with this primary profile"),
+            ignoreFocusOut: true,
+            canPickMany: false
+        };
+        chosenType = await vscode.window.showQuickPick(possibles, quickPickOptions1);
+        if (chosenType) {
+            const profiles = Profiles.getInstance().getNamesForType(chosenType);
+            const quickPickOptions2: vscode.QuickPickOptions = {
+                placeHolder: localize("profileLink.selectFileName", "Select the file name to associate with this primary profile"),
+                ignoreFocusOut: true,
+                canPickMany: false
+            };
+            chosenName = await vscode.window.showQuickPick(profiles, quickPickOptions2);
+            if (chosenName) {
+                try {
+                    await saveLinkedProfile(aProfile, chosenType, chosenName);
+                    vscode.window.showInformationMessage(localize("profileLink.associated", "Associated secondary profile {0}:{1} with {2}:{3} primary.",
+                                     chosenType,chosenName,aProfile.type,aProfile.name));
+                } catch (err) {
+                    vscode.window.showErrorMessage(localize("profileLink.unableToSave", "Unable to save profile association. ") + err.message);
+                }
+            }
+        }
+    }
+}
+
+async function findLinkedProfile(aProfile: IProfileLoaded, type: string) {
```

Comment:
From what I recall, we are trying to be more consistent with the blocking vs non-blocking methods. 
The direction (As far as I remember) was (wherever possible) try going with a non-blocking execution approach.
i.e. use `async-await` over nodejs sync methods.

@VitGottwald, @dkelosky , any thoughts

Reply:
We have async-await throughout. That's what VSCode recommends I believe.

Reply:
We can address this in a future issue though 😋